reporter/src/main/kotlin/reporters/NoticeByPackageReporter.kt

```diff
@@ -0,0 +1,190 @@
+                .filter { (license, _) -> license !in licensesInNoticeFiles }
+                .filter { (license, _) ->
+                    licenseTextProvider.hasLicenseText(license).also {
+                        if (!it) {
+                            log.warn {
+                                "No license text found for license '$license', it will be omitted from the report."
+                            }
+                        }
+                    }
+                }
+
+            if (processedFindings.isNotEmpty()) {
+                append("The following copyrights and licenses were found in the source code of this package:\n\n")
+            }
+
+            processedFindings.entries.forEachIndexed { index, (license, copyrights) ->
+                licenseTextProvider.getLicenseText(license)?.let { licenseText ->
+                    // Note: Do not use appendln() here as that would write out platform-native line endings, but we
+                    // want to normalize on Unix-style line endings for consistency.
+                    copyrights.forEach { copyright ->
+                        append("$copyright\n")
+                    }
+                    if (copyrights.isNotEmpty()) append("\n")
+
+                    append(licenseText)
+
+                    if (index < processedFindings.size - 1) {
+                        append(LICENSE_SEPARATOR)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun getFindingsForLicenseFiles(
+        scanResult: ScanResult,
+        archiveDir: File,
+        licenseFindingsMap: LicenseFindingsMap
+    ): MutableMap<String, LicenseFindingsMap> {
+        val matcher = FileMatcher(LICENSE_FILENAMES)
+        val licenseFiles = mutableMapOf<String, LicenseFindingsMap>()
+
+        archiveDir.walkTopDown().forEach { file ->
+            val relativePath = archiveDir.toPath().relativize(file.toPath())
+            if (matcher.matches(relativePath.toString())) {
+                val licenses = findLicensesForFile(scanResult, relativePath)
+                val copyrightsInFile = findCopyrightsForFile(scanResult, relativePath)
+
+                val licensesWithCopyrights = licenses.associateWith { license ->
+                    // Remove copyrights which are already contained in the license file.
+                    val allCopyrights = (licenseFindingsMap[license].orEmpty() + copyrightsInFile).toSet()
+                    val processedCopyrights = CopyrightStatementsProcessor().process(allCopyrights)
+                    val remainingProcessedCopyrights =
+                        processedCopyrights.processedStatements.filterNot { (_, sources) ->
+                            copyrightsInFile.any { it in sources }
+                        }
+                    val remainingUnprocessedCopyrights =
+                        processedCopyrights.unprocessedStatements - copyrightsInFile
```

Comment:
In conjunction with `remainingUnprocessedCopyrights` the name `processedCopyrights` might be a bit confusing.
Consider `processCopyrightsResult` instead so that you would end up with `processCopyrightsResult.processed...` and `processCopyrightsResult.unprocessed...

Reply:
Done.